File: skill_sdk/tests/fixtures/js-project/src/shop/payment/StripeGateway.js
import { PaymentGateway } from './PaymentGateway.js';
import { PaymentResult } from './PaymentResult.js';

/**
 * Stripe-backed payment gateway.
 *
 * @implements {PaymentGateway}
 */
export class StripeGateway extends PaymentGateway {
  /**
   * @param {string} apiKey
   */
  constructor(apiKey) {
    super();
    /** @type {string} */
    this._apiKey = apiKey;
  }

  /**
   * @param {number} amount
   * @param {string} currency
   * @returns {PaymentResult}
   */
  charge(amount, currency) {
    const txId = `stripe_ch_${this._hashFloat(amount)}`;
    return new PaymentResult(true, txId, `Charged ${amount} ${currency}`);
  }

  /**
   * @param {string} transactionId
   * @returns {PaymentResult}
   */
  refund(transactionId) {
    return new PaymentResult(
      true,
      `stripe_ref_${transactionId}`,
      `Refunded transaction ${transactionId}`
    );
  }

  /** @returns {string} */
  getApiKey() { return this._apiKey; }

  /**
   * @param {number} f
   * @returns {number}
   */
  _hashFloat(f) {
    return Math.floor(f * 100);
  }
}

File: skill_sdk/tests/fixtures/java-project/src/main/java/com/fixture/shop/payment/StripeGateway.java
package com.fixture.shop.payment;

/**
 * Stripe-backed payment gateway.
 */
public class StripeGateway implements PaymentGateway {

    private final String apiKey;

    public StripeGateway(String apiKey) {
        this.apiKey = apiKey;
    }

    @Override
    public PaymentResult charge(double amount, String currency) {
        String txId = "stripe_ch_" + hashFloat(amount);
        return new PaymentResult(true, txId, "Charged " + amount + " " + currency);
    }

    @Override
    public PaymentResult refund(String transactionId) {
        return new PaymentResult(true, "stripe_ref_" + transactionId,
                "Refunded transaction " + transactionId);
    }

    public String getApiKey() {
        return apiKey;
    }

    private static int hashFloat(double f) {
        return (int) (f * 100);
    }
}
